aac_codec_registration.src.html

```diff
@@ -0,0 +1,178 @@
+Repository: w3c/webcodecs
+Status: NOTE-ED
+Shortname: webcodecs-aac-codec-registration
+Level: none
+Group: mediawg
+ED: https://w3c.github.io/webcodecs/aac_codec_registration.html
+TR: https://www.w3.org/TR/webcodecs-aac-codec-registration/
+Editor: Chris Cunningham, w3cid 114832, Google Inc. https://www.google.com/
+Editor: Paul Adenot, w3cid 62410, Mozilla https://www.mozilla.org/
+Editor: Bernard Aboba, w3cid 65611, Microsoft Corporation https://www.microsoft.com/
+
+Abstract: This registration is entered into the [[webcodecs-codec-registry]].
+    It describes, for AAC, (1) the fully qualified codec strings, (2)
+    the {{AudioDecoderConfig.description}} bytes, (3) the codec-specific
+    {{EncodedAudioChunk}} [=EncodedAudioChunk/[[internal data]]=] bytes, and (4)
+    the values of {{EncodedAudioChunk}} [=EncodedAudioChunk/[[type]]=].
+
+    The registration is not intended to include any information on whether a
+    codec format is encumbered by intellectual property claims. Implementers and
+    authors are advised to seek appropriate legal counsel in this matter if they
+    intend to implement or use a specific codec format. Implementers of
+    WebCodecs are not required to support the AAC codec.
+
+    This registration is non-normative.
+
+Markup Shorthands:css no, markdown yes, dfn yes
+!Participate: <a href="https://github.com/w3c/webcodecs">Git Repository.</a>
+!Participate: <a href="https://github.com/w3c/webcodecs/issues/new">File an issue.</a>
+!Version History: <a href="https://github.com/w3c/webcodecs/commits">https://github.com/w3c/webcodecs/commits</a>
+</pre>
+
+<pre class='anchors'>
+spec: WEBCODECS; urlPrefix: https://w3c.github.io/webcodecs/#
+    type: attribute
+        text: AudioDecoderConfig.description; url: dom-audiodecoderconfig-description
+        text: AudioDecoderConfig.sampleRate; url: dom-audiodecoderconfig-samplerate
+        text: AudioDecoderConfig.channelCount; url: dom-audiodecoderconfig-channelcount
+    type: dfn
+        for: EncodedAudioChunkType; text: key; url: dom-encodedaudiochunktype-key
+        for: EncodedAudioChunk; text: [[internal data]]; url: dom-encodedaudiochunk-internal-data-slot
+        for: EncodedAudioChunk; text: [[type]]; url: dom-encodedaudiochunk-type-slot
+    type: interface
+        text: EncodedAudioChunk; url: encodedaudiochunk
+    type: dictionary
+        text: AudioDecoderConfig; url: dictdef-audiodecoderconfig
+        text: AudioEncoderConfig; url: dictdef-audioencoderconfig
+</pre>
+
+<pre class='biblio'>
+{
+  "iso14496-3": {
+    "href": "https://www.iso.org/standard/53943.html",
+    "title": "ISO/IEC 14496-3:2009 - Information technology — Coding of audio-visual objects — Part 3: Audio",
+    "publisher": "ISO",
+    "date": "2009-09"
+  }
+}
+</pre>
+
+Fully qualified codec strings {#fully-qualified-codec-strings}
+==============================================================
+
+This codec has multiple possible codec string:
+
+- `"mp4a.40.2"`
```

Comment:
Can you offer some description of the AAC profiles described by the different strings (and which strings are simply duplicates for historical reasons). E.g. mp4a.40.2
```
      //   mp4a.40.2   - MPEG-4 AAC LC
      //   mp4a.40.02  - MPEG-4 AAC LC (leading 0 in aud-oti for
      //                 compatibility)
      //   mp4a.40.5   - MPEG-4 HE-AAC v1 (AAC LC + SBR)
      //   mp4a.40.05  - MPEG-4 HE-AAC v1 (AAC LC + SBR) (leading 0
      //                 in aud-oti for compatibility)
      //   mp4a.40.29  - MPEG-4 HE-AAC v2 (AAC LC + SBR + PS)
```

Chrome also recognizes these MPEG2 strings.
```
      //   mp4a.66     - MPEG-2 AAC MAIN
      //   mp4a.67     - MPEG-2 AAC LC
      //   mp4a.68     - MPEG-2 AAC SSR
```

I'm not an expert on these, but Wikipedia suggests that the LC of mpeg-4 may also include "Perceptual Noise Substitution (PNS)", not found in that of mpeg-2. Maybe that's a meaningful distinct from mpeg4 LC above (particularly if you have some ancient mpeg-2 codec). Our decode pipeline just sends it all to ffmpeg.

Reply:
Here's Gecko list: https://searchfox.org/mozilla-central/source/dom/media/VideoUtils.cpp#664-673, it only has `"mp4a.67"` from this second list. AFAIK (I just searched our bugzilla), we haven't had reports of this being a problem, did Chromium hear anything about those?

Not that I'm opposed about adding more though.

Reply:
I'm not sure on the history of the MPEG-2 strings. We don't have good metrics to say what current usage is like.  @dalecurtis, any reason we shouldn't register these? Do we mind their continued usage?

@padenot does Gecko generally decode AAC w/ FFmpeg? If so, you may actually support these even without having advertised it. It could be that this is mostly used on sites that just do <audio src=> without bothering to canPlayType().

Reply:
> @padenot does Gecko generally decode AAC w/ FFmpeg? If so, you may actually support these even without having advertised it. It could be that this is mostly used on sites that just do <audio src=> without bothering to canPlayType().

Only on Linux, if `ffmpeg` has been installed on the host. On macOS, Windows and Android, the OS decoders are used.

Reply:
These all come from here: https://mp4ra.org/#/object_types -- sticking with only the mp4a.40.xx ones seems reasonable to me in this instance, but whatever is easiest.